src/Inciter/AMR/Error.C

```diff
@@ -0,0 +1,119 @@
+// *****************************************************************************
+/*!
+  \file      src/Inciter/AMR/Error.C
+  \copyright 2012-2015, J. Bakosi, 2016-2017, Los Alamos National Security, LLC.
+  \brief     Class for computing error estimates for mesh refinement
+  \details   Class for computing error estimates for mesh refinement.
+*/
+// *****************************************************************************
+
+#include <cmath>
+#include <limits>
+
+#include "Exception.h"
+#include "Error.h"
+#include "Vector.h"
+#include "Gradients.h"
+#include "Inciter/Options/AMRError.h"
+
+using AMR::Error;
+
+tk::real
+Error::scalar( const tk::Fields& u,
+               const std::pair< std::size_t, std::size_t >& edge,
+               ncomp_t c,
+               const std::array< std::vector< tk::real >, 3 >& coord,
+               const std::vector< std::size_t >& inpoel,
+               const std::pair< std::vector< std::size_t >,
+                                std::vector< std::size_t > >& esup,
+               inciter::ctr::AMRErrorType err )
+// *****************************************************************************
+//  Estimate error for scalar quantity
+//! \param[in] u Solution vector
+//! \param[in] edge Edge defined by its two end-point IDs
+//! \param[in] c Scalar component to compute error of
+//! \param[in] coord Mesh node coordinates
+//! \param[in] inpoel Mesh element connectivity
+//! \param[in] esup Linked lists storing elements surrounding points, see
+//!    tk::genEsup()
+//! \param[in] err AMR Error indicator type
+//! \return Error indicator: a real number between [0...1] inclusive
+// *****************************************************************************
+{
+  if (err == inciter::ctr::AMRErrorType::JUMP)
```

Comment:
Not a big deal, but could we find a way to remove the branching/run time check here?

Reply:
What do you mean? Should it be compile-time instead? Or some other way?

Reply:
Okay, We could eliminate the runtime branching here if we trigger off the error indicator type from what the user configured in the input file. I didn't do that originally, because this way this `Error` class is independent of the user input (`g_inputdeck`), but I guess we it is okay to use it.